seeding acm agent

bjoydeep opened the pull request:
Checking in the:

1. notebook that can handle creating policy as well as doing a search query.

> Disclaimer - the Pull request and scorer agent is not done. It is a place holder. 
> Idea is that @yanmxa can add the other agents he is already working on - that can do kubectl/cluster-proxy calls etc here asap.

2. Likewise, checking in simple chat interface src/acm_chat.py that can execute the same thing as a CLI

> Main idea again is to speed up creation of a acm agent.

A simple outcome of the fully functioning agent would be -

1. user can find details about policy from docs - we can add a simple doc lookup agent 
2. user can then create a policy and placement etc and generate a PR. 
4. user can then query status about the policy  - from search or if details are not in search it will query kubectl on hub or through cluster-proxy
5. The agent should be able to handle comments in PR as well
6. user can query prom/thanos metrics as well.

While we are doing this, we also start to migrate into llama stack and MCP.

Question -
There is existing code in this repo which is old - and will probably not work anymore or will need to be upgraded. But these are very early code, so may not be very relevant anymore. What do we do with those? I also do not want to totally delete the knowledge graph work. WDYT?

bjoydeep committed:
seeding acm agent

Signed-off-by: Joydeep Banerjee <jbanerje@redhat.com>

yanmxa commented:
1. We can also add a subdirectory named `archived` to ensure these old code exists in the current project. 
2. I see that your prompt about GRC is a good start for embedding them into the MCP server as the pre-written template: https://modelcontextprotocol.io/docs/concepts/prompts

yanmxa commented on src/acm_agents.py:
```diff
@@ -0,0 +1,268 @@
+#gpt-3.5-turbo
+#o1
+#o3-mini
+model = ChatOpenAI(model="gpt-4o", temperature=0)
```
Can the demo run with the groq client/model? 

I'd love to try it with your workflow - it's really attractive!

bjoydeep committed:
moving old files to archive

Signed-off-by: Joydeep Banerjee <jbanerje@redhat.com>

bjoydeep commented:
Hi @yanmxa  -  I will add groq in the next commit. As you mentioned earlier -

1. moved old stuff into archive
2. made the database work for remote DB
3. made some minor changes in the launcher so that the command line experience is a little better 
4. added some comments to the code.

yanmxa commented:
/lgtm
